akka-http-core/src/main/scala/akka/http/impl/engine/client/OutgoingConnectionBlueprint.scala

```diff
@@ -205,14 +204,8 @@ private[http] object OutgoingConnectionBlueprint {
       }
 
       override def onDownstreamFinish(): Unit = {
-        // if downstream cancels while streaming entity,
-        // make sure we also cancel the entity source, but
-        // after being done with streaming the entity
-        if (entitySubstreamStarted) {
-          completionDeferred = true
-        } else {
-          completeStage()
-        }
+        if (entitySubstreamStarted) entitySource.complete()
```

Comment:
question: complete, as we do nowadays, or is this actually a fail... see the previous commit in this PR (completing is the current semantics expected in other tests)